src/libraries/System.IO.Ports/src/System/IO/Ports/SerialPort.Unix.cs

```diff
@@ -5,13 +5,28 @@
 using System.ComponentModel;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.InteropServices;
 
 namespace System.IO.Ports
 {
     public partial class SerialPort : Component
     {
         public static string[] GetPortNames()
         {
+#if NETCOREAPP
+            return OperatingSystem.IsLinux() ? GetPortNames_Linux()
+                : OperatingSystem.IsMacOS() ? GetPortNames_OSX()
+                : OperatingSystem.IsFreeBSD() ? GetPortNames_FreeBSD()
+#else
+            return RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? GetPortNames_Linux()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ? GetPortNames_OSX()
+                : RuntimeInformation.IsOSPlatform(OSPlatform.Create("FREEBSD")) ? GetPortNames_FreeBSD()
+#endif
+                : throw new PlatformNotSupportedException(SR.PlatformNotSupported_SerialPort_GetPortNames);
```

Comment:
@wfurt should we fallback here?

Reply:
we may return empty list. But I like the `throw` as it makes it clear if anybody is trying to get this working on new platform. And we don't build other platforms anyway so it would still need some work.